src/silo/query_engine/actions/mutations.cpp

```diff
@@ -63,35 +63,92 @@ std::unordered_map<std::string, typename Mutations<SymbolType>::PrefilteredBitma
+template <typename SymbolType>
+void Mutations<SymbolType>::addPositionToMutationCountsForFullBitmaps(
+   uint32_t position,
+   const PrefilteredBitmaps& bitmaps_to_evaluate,
+   SymbolMap<SymbolType, std::vector<uint32_t>>& count_of_mutations_per_position
+) {
    // For these partitions, we have full bitmaps. Do not need to bother with AND
    // cardinality
    for (const auto& [filter, sequence_store_partition] : bitmaps_to_evaluate.full_bitmaps) {
-      for (const auto symbol : SymbolType::VALID_MUTATION_SYMBOLS) {
-         if (sequence_store_partition.positions[position].symbol_whose_bitmap_is_flipped != symbol) {
-            count_of_mutations_per_position[symbol][position] +=
-               sequence_store_partition.positions[position].bitmaps.at(symbol).cardinality();
-         } else {
+      for (const auto symbol : SymbolType::SYMBOLS) {
+         const auto& current_position = sequence_store_partition.positions[position];
+         if (current_position.isSymbolDeleted(symbol)) {
             count_of_mutations_per_position[symbol][position] +=
-               sequence_store_partition.sequence_count -
-               sequence_store_partition.positions[position].bitmaps.at(symbol).cardinality();
+               sequence_store_partition.sequence_count;
+            continue;
+         }
+         const uint32_t symbol_count = current_position.isSymbolFlipped(symbol)
+                                          ? sequence_store_partition.sequence_count -
+                                               current_position.getBitmap(symbol)->cardinality()
+                                          : current_position.getBitmap(symbol)->cardinality();
+
+         count_of_mutations_per_position[symbol][position] += symbol_count;
+
+         const auto deleted_symbol = current_position.getDeletedSymbol();
+         if (deleted_symbol.has_value() && symbol != *deleted_symbol) {
+            count_of_mutations_per_position[*deleted_symbol][position] -=
+               sequence_store_partition.positions[position].getBitmap(symbol)->cardinality();
+         }
+      }
+   }
+}
+
+template <typename SymbolType>
+void Mutations<SymbolType>::correctDeletedSymbolCountForMissingSymbols(
+   uint32_t position,
+   const typename silo::query_engine::actions::Mutations<SymbolType>::PrefilteredBitmaps&
+      bitmaps_to_evaluate,
+   SymbolMap<SymbolType, std::vector<uint32_t>>& count_of_mutations_per_position
+) {
+   for (const auto& [filter, sequence_store_partition] : bitmaps_to_evaluate.bitmaps) {
+      const auto deleted_symbol = sequence_store_partition.positions[position].getDeletedSymbol();
+      if (deleted_symbol == std::nullopt) {
+         continue;
+      }
+      for (const uint32_t idx : *filter) {
+         const roaring::Roaring& n_bitmap = sequence_store_partition.missing_symbol_bitmaps[idx];
+         if (n_bitmap.contains(position)) {
+            count_of_mutations_per_position[*deleted_symbol][position] -= 1;
+         }
+      }
```

Comment:
As discussed, let's inline this into `addPositionToMutationCountsForMixedBitmaps` and get rid of the separate `correct...` method.